pkg/controller/disruption/disruption.go

```diff
@@ -647,13 +654,14 @@ func (dc *DisruptionController) getExpectedScale(pdb *policy.PodDisruptionBudget
 	for _, pod := range pods {
 		controllerRef := metav1.GetControllerOf(pod)
 		if controllerRef == nil {
-			err = fmt.Errorf("found no controller ref for pod %q", pod.Name)
-			dc.recorder.Event(pdb, v1.EventTypeWarning, "NoControllerRef", err.Error())
-			return
+			dc.recorder.Event(pdb, v1.EventTypeWarning, "NoControllerRef",
+				fmt.Sprintf("found no controller ref for pod %q", pod.Name))
+			continue
```

Comment:
is getExpectedScale only used in percentage cases? our docs on uncontrolled pods (https://kubernetes.io/docs/tasks/run-application/configure-pdb/#arbitrary-controllers-and-selectors) don't mention PDB is unusable for those pods, only that percentages cannot be used.

Is there a reason we wouldn't treat uncontrolled pods or pods belonging to a controller that does not implement `/scale` as having an implicit expected scale of 1?

Reply:
I also wonder if `minAvailable: "100%"`, `maxUnavailable: "0%"`, `maxUnavailable: 0` should also be specially handled, since they serve to effectively block all evictions

Reply:
So I think we could make these special cases and just set `allowedDisruptions` to 0 if we encounter any of them. But I'm not sure if we should. 
If there happens to be more pods than specified by the scale of the controller(s), we probably should allow a disruption. This is probably a rare situation and something the workload controllers will clean up, but it seems like we should think about it. 
Also, if we are considering "short-circuiting" the reconcile loop in these situation and not look at the pods and their controllers at all, we will not have the information needed in the other properties in the PDB status object. These are not used by the Eviction API, but I think we should still try to keep the information correct here.

Reply:
`getExpectedScale` is only called when `maxUnavailable` is used or if `minAvailable` is used with a percentage. I think this is covered by the documentation through the two criteria listed.

I'm not sure it would provide any value if we allow uncontrolled pods to be used with `maxUnavailable` or `minAvailable` with a percentage. We would essentially compute the desired scale to be the current number of pods. A minAvailable of `50%` would mean that half the pods will always be available for disruption, not matter how many pods are actually running. If I have 10 pods, I will be able to terminate 5 of them. Once that is done, I will be able to terminate another 2 and so on.
It might provide some protection against terminating pods when some of them are not in the `Ready` condition. We could calculate the scale based on all pods (even if they are not `Ready`) while we obviously will only use `Ready` pods for computing the number of actually available pods. I'm a little worried that this will lead to some confusing behavior.